enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,882 @@
+has access to any secrets in the cluster and builds the credentials to access
+the registries consuming the inspected pod's `imagePullSecrets` field. The
+controller will also read the image content source policy objects once the
+support for restricted network scenarios are implemented.
+If any unrecoverable error occurs during the image inspection process,
+the scheduling gate is removed from the pod, and no changes are made to the pod's
+affinity.
+This logic prevents the cluster's workloads from being indefinitely held
+before entering the ActiveQ scheduling queue.
+
+#### Required RBAC permissions
+
+The service account running the operator components will need the following RBAC
+permissions:
+
+- Deployment (namespace-scoped)
+  - Watch, List, Get, Create, Update, Delete
+- Pod (cluster-scoped)
+  - Watch, List, Get, Update
+- Secrets (cluster-scoped)
+  - Watch, List, Get
+- ConfigMaps (MMO operator namespace and openshift-image-registry)
+  - Watch, List, Get
+- PodPlacementConfig CRD
+  - Watch, List, Get, Update
+- admissionregistration.k8s.io/MutatingWebhookConfiguration
+  - Get, Update
+- image.openshift.io/ImagesStream/layers (required for internal registry access)
+  - Get
+- operator.openshift.io/ImageContentSourcePolicy
+- config.openshift.io/ImageDigestMirrorSet
+- config.openshift.io/ImageTagMirrorSet
+  - Watch, List, Get
+- config.openshift.io/Image
+  - Watch, List, Get
+
+#### PodPlacementConfig CR
+The configuration CR of the operator allows the admins to set the following:
+- a `NamespaceSelector` to filter the namespaces MMO should patch the pods of.
+The `openshift-*`, `kube-*` and `hypershift-*` namespaces are implicitly excluded.
+- the `LogVerbosity` level of the operator
+A new Custom Resource Definition (CRD) is being proposed to handle the
+configuration of the operator deployment.
+
+```go
+// +kubebuilder:validation:Enum=Normal;Debug;Trace;TraceAll
+type LogVerbosityLevel string
+
+const (
+    LogVerbosityLevelNormal LogVerbosityLevel = "Normal"
+    LogVerbosityLevelDebug LogVerbosityLevel = "Debug"
+    LogVerbosityLevelTrace LogVerbosityLevel = "Trace"
+    LogVerbosityLevelTraceAll LogVerbosityLevel = "TraceAll"
+)
+
+
+// PodPlacementConfigSpec defines the desired state of PodPlacementConfig
+type PodPlacementConfigSpec struct {
+    // LogVerbosity is the log level for the pod placement controller
+    // Valid values are: "Normal", "Debug", "Trace", "TraceAll".
+    // Defaults to "Normal".
+    // +optional
+    // +kubebuilder:default=Normal
+    LogVerbosity LogVerbosityLevel `json:"logVerbosity,omitempty"`
+
+    // NamespaceSelector is a metav1.LabelSelector to filter the namespaces whose
+    // pods should be patched with the scheduling gate at admission time.
+    // All the pods in the excluded namespaces will not be controlled by the operator.
+    // The webhook will always ignore pods in reserved namespaces matching the
+    // openshift-*, kube-*, and hypershift-* glob.
```

Comment:
Nits, think it might be clearer worded this way
```suggestion
    // namespaceSelector is a metav1.LabelSelector to filter the namespaces whose
    // pods should be patched with the scheduling gate at admission time.
    // Any pod in a namespace not matched by the selector will not be controlled by the operator.
    // The webhook will always ignore pods in reserved namespaces matching the
    // openshift-*, kube-*, and hypershift-* globs.
```